nntrainer/graph/network_graph.cpp

```diff
@@ -490,4 +496,43 @@ int NetworkGraph::setEdge() {
   return status;
 }
 
+void NetworkGraph::setBatchSize(unsigned int batch_size) {
+  for (auto const &layer_node : Sorted) {
+    layer_node.layer->setBatch(batch_size);
+  }
+}
+
+sharedConstTensors NetworkGraph::forwarding(sharedConstTensors input) {
+  for (unsigned int i = 0; i < Sorted.size() - 1; ++i) {
+    LayerNode &layer_node = Sorted[i];
+    if (layer_node.layer->getType() == LayerType::LAYER_IN) {
+      layer_node.layer->forwarding(input);
+    } else {
+      layer_node.layer->forwarding();
+    }
+  }
+
+  std::vector<sharedConstTensor> out;
+
+  for (unsigned int i = 0; i < Sorted[Sorted.size() - 2].layer->num_outputs;
+       ++i) {
+    out.push_back(
+      MAKE_SHARED_TENSOR(Sorted[Sorted.size() - 2].layer->net_hidden[i]->var));
+  }
+
+  return out;
+}
+
+void NetworkGraph::backwarding(sharedConstTensors output, int iteration) {
+
+  for (unsigned int i = Sorted.size() - 1; i > 0; i--) {
+    LayerNode &layer_node = Sorted[i];
+    if (layer_node.layer->getType() == LayerType::LAYER_LOSS) {
+      layer_node.layer->backwarding(iteration, output);
```

Comment:
same, why LAYER_LOSS not following the same semantics as other LAYERS?

Reply:
There is two kind of forwarding. one is to calculate the loss and without calculation loss. In order to calculate the loss, it just take input (which is return from previous layer) and calculate old fashion way. I though it's not a big deal. We can change this later pr though.